contracts/ERC721A.sol

```diff
@@ -30,6 +30,11 @@ interface ERC721A__IERC721Receiver {
  * Assumes that the maximum token id cannot exceed 2**256 - 1 (max value of uint256).
  */
 contract ERC721A is IERC721A {
+    // Reference type for token approval.
+    struct TokenApprovalRef {
+        address value;
+    }
```

Comment:
I like to mention that we could move this to the interface, but for private use, I think it's ok to be here.

Reply:
Definitely only for private use. 

Gave it to a more obscure name like `TokenApprovalRef` instead of `TokenApproval` to prevent namespace collision.